crates/cli/tests/cli_run.rs

```diff
@@ -821,25 +821,6 @@ mod cli_run {
             if entry.file_type().unwrap().is_dir() {
                 let example_dir_name = entry.file_name().into_string().unwrap();
 
-                // TODO: Improve this with a more-dynamic approach. (Read all subdirectories?)
-                // Some platform-switching examples live in nested directories
-                if example_dir_name == "platform-switching" {
-                    for sub_dir in [
-                        // We exclude the C platforming switching example
-                        // because the main platform switching example runs the c platform.
-                        // If we don't a race condition leads to test flakiness.
-                        // "c-platform",
-                        "rust-platform",
-                        "swift-platform",
-                        "web-assembly-platform",
-                        "zig-platform",
-                    ] {
-                        all_examples.remove(format!("{}/{}", example_dir_name, sub_dir).as_str()).unwrap_or_else(|| {
-                            panic!("The example directory {}/{}/{} does not have any corresponding tests in cli_run. Please add one, so if it ever stops working, we'll know about it right away!", examples_dir, example_dir_name, sub_dir);
-                        });
-                    }
-                }
-
                 // We test benchmarks separately
                 if example_dir_name != "benchmarks" {
```

Comment:
I'm expecting that we'll need to add a matching "platform-switching" check to this line, but let's see what the workflows say...

Reply:
I thought so too at first. I think this is what's happening:

If we add `&& example_dir_name != "platform-switching"` to the if statement, the platform switching tests no longer get removed from the collection of tests. Therefore, the assertion that the collection is empty at the end fails. It is ok to ignore the benchmarks, because benchmarks weren't added to the collection in the first place.

At least that's my best guess.

Reply:
Huh! Well, the tests pass, so that's good enough for me. Thank you, @sandprickle! I'm happy to pair with you on your next endeavor, too (hopefully with fewer deep surprises, haha)